core/src/main/scala/za/co/absa/spline/core/harvester/DataLineageBuilder.scala

```diff
@@ -62,11 +75,20 @@ class DataLineageBuilder(logicalPlan: LogicalPlan, executedPlanOpt: Option[Spark
           if (parentBuilder != null) parentBuilder += curBuilder
 
           if (maybeExistingBuilder.isEmpty) {
-            val newNodesToProcess =
-              writeCommandParser.
+
+            //try to find all possible commands for traversing- save to filesystem, saveAsTable, JDBC
+            val writes = writeCommandParser.
                 asWriteCommandIfPossible(curOpNode).
                 map(wc => Seq(wc.query)).
-                getOrElse(curOpNode.children)
+                getOrElse(Nil)
+
+            val tables = tableCommandParser.
```

Comment:
Maybe we could have single WriteNode builder that would find both table writes and file writes. If not then I would rename writeCommand to fileWriteCommand or similar.